libcontainer/container_linux.go

```diff
@@ -695,13 +683,25 @@ func (c *linuxContainer) newInitConfig(process *Process) *initConfig {
 	return cfg
 }
 
-func (c *linuxContainer) Destroy() error {
+// Destroys the container, if its in a valid state, after killing any
+// remaining running processes.
+//
+// Any event registrations are removed before the container is destroyed.
+// No error is returned if the container is already destroyed.
+//
+// Running containers must first be stopped using Signal(..).
+// Paused containers must first be resumed using Resume(..).
+func (c *Container) Destroy() error {
 	c.m.Lock()
 	defer c.m.Unlock()
 	return c.state.destroy()
 }
 
-func (c *linuxContainer) Pause() error {
+// If the Container state is RUNNING or CREATED, sets the Container
+// state to PAUSING and pauses the execution of any user processes.
```

Comment:
```suggestion
// Pause sets the Container state to PAUSING and pauses the execution of any
// user processes if the Container state is RUNNING or CREATED.
```

Reply:
What this commit did is 

> Remove BaseContainer and Container interfaces, moving their methods
> documentation to linuxContainer.

I was not changing any comments.

Also, it seems that the doc is wrong and the async part and "pausing" state was removed long time ago (I was not able to find it being used in git history).

Let's address this separately (on top of this PR once it's merged). Opened https://github.com/opencontainers/runc/issues/3343 so we won't forget.